src/m3ninx/index/segment/builder/builder.go

```diff
@@ -236,7 +298,9 @@ func (b *builder) Fields() (segment.FieldsIterator, error) {
 }
 
 func (b *builder) Terms(field []byte) (segment.TermsIterator, error) {
-	terms, ok := b.fields.Get(field)
+	// NB(bodu): The # of indexQueues and field map shards are equal.
+	shard := int(xxhash.Sum64(field) % uint64(len(b.indexQueues)))
```

Comment:
Can we make this a method on the builder to reuse this code?  I see it replicated on this line and line 230 i.e.

```go
func (b *builder) shardField(field []byte) {
  return int(xxhash.Sum64(field) % uint64(len(b.indexQueues)))
}
```